WooCommerce/UITestsFoundation/Screens/Products/ProductsScreen.swift

```diff
@@ -59,14 +48,17 @@ public final class ProductsScreen: ScreenObject {
         return try SingleProductScreen()
     }
 
-    public func verifyProductListOnProductsScreen(count: Int, name: String, status: String) throws -> Self {
-        XCTAssertTrue(try app.getTextVisibilityCount(text: name) == 1, "Product name does not exist!")
-        XCTAssertTrue(app.tables.cells.count == count, "Expecting \(count) products, got \(app.tables.cells.count) instead!")
+    func verifyStockStatusForProduct(name: String, status: String) throws -> Bool {
+        let namePredicate = NSPredicate(format: "identifier == %@", name)
+        let statusPredicate = NSPredicate(format: "label CONTAINS[c] %@", status)
```

Comment:
We know all the possible stock status values, so I think this will be more robust:

```suggestion
        let statusPredicate = NSPredicate(format: "label == %@", status)
```

(you'll need to capitalize the first letter of the stock statuses dictionary values)

Reply:
Ah ok good point, but I think it's better to keep it all lowercase so we don't have to case about the case, I'll go with this `let statusPredicate = NSPredicate(format: "label ==[c] %@", status)` which I've tested also works